CHANGELOG.md

```diff
@@ -2,6 +2,110 @@
 
 This contains only the most important and/or user-facing changes; for a full changelog, see the commit history.
 
+## [2.0.0](https://github.com/ably/ably-js/tree/2.0.0) (date TBD)
+
+The 2.0.0 release introduces a number of new features and QoL improvements, including a new way to remove bloat and reduce the bundle size of your ably-js client, first-class support for Promises, a more idiomatic approach to using ably-js' React Hooks, enhancements to TypeScript typings, and more.
+
+Below is a list of all new features, APIs, deprecations, and breaking changes.
```

Comment:
I don't think that overlap by itself is a bad thing.
Also, as it currently stands, the migration guide is very "raw" and goes immediately into the code details on how to migrate to v2, which I believe it should do.
The changelog lists all the things that were added, changed, and improved, while the migration guide explains step by step how to migrate. So, I don't think there is a real overlap here.

> highlight the most important breaking changes (e.g. platform compatibility) and direct users to the migration guide for the full details?

Changelog already has a list of most important features listed in separate headings, and the link to the migration guide is on the line below. Only after that we list all of the PRs one by one.
Did you have something else in mind on how to do this?
The changelog already lists the most important features under separate headings below (before listing PRs), and the link to the migration guide is provided on the line below.
Did you have something else in mind on how to do this?

Reply:
> The changelog lists all the things that were added, changed, and improved

Maybe it would be good to organise the prose part of the changelog into sections that align with these categories?

> I don't think that overlap by itself is a bad thing.

Agreed, I think I was not very careful in my comment before. The aim is to have a set of documentation that's useful to users, and that might involve repeating ourselves.

Reply:
> Maybe it would be good to organise the prose part of the changelog into sections that align with these categories?

If by `prose part` you mean those high-level topics like "Bundle Size Reduction" and "React Hooks", I feel like it doesn't really make sense for them. It's better to list them as is, by part of the application that was impacted and explain what's changed for each of them.

For the PRs though, I reordered them so they form those categories.

Reply:
My thinking was that it would be nice for a user who looks at the changelog to know which stuff they _need_ to pay attention to (because else their code will break) and which stuff they could read later. But up to you.

Reply:
Reordered PRs in https://github.com/ably/ably-js/pull/1656/commits/f0995b4792a7dbcb73dd4e19a22412f8ee90b073